### PR TITLE
refactor(tests): move tests to separate files and conftest.py for fixtures

### LINKED ISSUE #57: Refactoring of tests - move command tests to separate files
The tests are getting a bit overwhelming as everything is scrammed into one file. 

A first step would be to move out single commands into a commands folder and make a seperate `test_command_<name>.py` files.

### PR DESCRIPTION
**Pull request by scratchmex**
I tried fix #57 by moving the tests into separate files and creating a `conftest.py` for fixtures.

Feedback welcome

PS: Happy hacktoberfest

### CODE REVIEW

**Comment by pmayd** [comment]
tests are normally outside of the Python package and I want to keep it this way. There is no need to ship the tests as a user will never run the tests. Only developers run the tests and they work on the source repository.

**Comment by pmayd** [comment]
And of course src keeps src there is a good reason for using src and not another top level dir

**Review by pmayd — CHANGES_REQUESTED** [review]
Please leave the overall structure of the project unchanged, so `src` remains where it is and `tests` also stay outside of the package

**Review by pmayd — CHANGES_REQUESTED** [review]
And please change the target branch to develop, we never change the master branch per PR

**Comment by pogross** [comment]
As mentioned above please
- use the develop branch
- don't change the overall structure of the project, #57 is related to refactoring within the tests folder

**Comment by scratchmex** [comment]
Now I just moved the tests into multiple files. Take a look

**Comment by scratchmex** [comment]
Rebased develop branch to solve merge problems.

File: tests/test_core.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import pytest
```
Comment (pmayd) [inline tests/test_core.py:1]: pytest is unused here

**Review by pmayd — CHANGES_REQUESTED** [review]
Looks good, nice separation and good use of conftest.py, didn't know about it.

However, tests fail because you did not run the pre-commit hooks. Please read the email and install pre-commit, then install it and run it with `pre-commit run --all-files`. This will fix some things like wrong order of imports and linting stuff. IF everything is green, you can commit the files and push again and we can merge this!

File: tests/test_karma.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    )
+
+    karma = Karma("ABC123", KARMABOT_ID, "CHANNEL42")
+    assert (
+        karma.change_karma(-3)
+        == "Not cool pybob lowering my karma to 12, but you are probably right, I will work harder next time"
+    )
+
+
+def test_process_karma_changes():
```
Comment (pmayd) [inline tests/test_karma.py:79]: Remove all empty test functions pls

File: tests/test_messages.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Messages / Slack
+import pytest
+
+from karmabot.slack import (
+    perform_text_replacements,
+)
+
+
+def test_get_cmd():
```
Comment (pmayd) [inline tests/test_messages.py:9]: again you can delete all empty tests

File: tests/test_karma.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    pass
+
+
+def _channel_score(channel):
```
Comment (pmayd) [inline tests/test_karma.py:83]: this belongs to the command topchannels, I suggest creating a new file test_karma_commands.py and move it there, along with test_channel_score

File: tests/test_karma.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            channel_info["name"],
+            channel_info["purpose"]["value"],
+            len(channel_info["members"]),
+            float(channel_info["latest"]["ts"]),
+            channel_info["latest"].get("subtype"),
+        )
+    )
+
+
+def test_channel_score(mock_slack_api_call, frozen_now):
```
Comment (pmayd) [inline tests/test_karma.py:97]: move to test_karma_commands.py

File: tests/test_karma.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        ("neutral", "neutral"),
+        ("chuck", "chuck"),
+        ("", "all"),
+        ("al", "all"),
+        ("neutr", "neutral"),
+        ("chuk", "chuck"),
+        ("help", "all"),
+    ],
+)
+def test_get_closest_category(user_category, expected):
```
Comment (pmayd) [inline tests/test_karma.py:125]: move to test_karma_commands.py, belongs to the joke command

**Review by pmayd — CHANGES_REQUESTED** [review]
Please delete all empty test functions and move the mentioned functions into a new file test_karma_commands.py

**Comment by pogross** [comment]
Also, please start clean from current develop branch and add your changes there. So we are sure everything works and we have not any problems due to the changing of branches and force pushes

File: tests/test_core.py (hunk 1 above)
Comment (scratchmex) [reply tests/test_core.py:1]: I left there on purpose. It's just an habit to import pytest on test files. If you mind that much I can remove it

**Comment by scratchmex** [comment]
> 
> 
> Also, please start clean from current develop branch and add your changes there. So we are sure everything works and we have not any problems due to the changing of branches and force pushes

My master is your develop, I rebased to avoid problems as I said. I did that to avoid opening another PR.

**Comment by scratchmex** [comment]
All suggestions done @pmayd . Let me know what you think.

File: tests/test_karma.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
```
Comment (pmayd) [inline tests/test_karma.py:89]: this function also belongs in test-karma_commands.py, then we are good to go"

**Review by pmayd — CHANGES_REQUESTED** [review]
Please move this one function over to karma_commands and we can merge!

**Comment by pmayd** [comment]
After fixing the last issue, we are good to merge!

**Review by pmayd — APPROVED** [review]
ready to be merged

**Review by pmayd — APPROVED** [review]
ready to be merged!

**Review by pogross — APPROVED** [review]
Good work! 👍 ShipIt

**Comment by pogross** [comment]
This closes #57

**Pull request merged**